### DISCUSSION TITLE
Output String?

### DISCUSSION THREAD

**Discussion by chrismalek** [maragudk/gomponents#97] [Q&A] [answered]
I may be missing something obvious. I am working with a framework that just needs a string for the HTML output. Is there a method / function / trick to do that with a node?

**Comment by markuswustenberg** [comment, marked as answer]
Every `Node` has a `Render` method, which you can pass any writer to. A good writer could be `strings.Builder` for you purpose. 😊

**Comment by chrismalek** [comment]
Great thanks. I end up doing something like this:

```go

	buf := new(bytes.Buffer)

       // gocomponent node generated "above the fold".  
	personHTML.Render(buf)
       // Iris web framework needs a string 
	ctx.Negotiation().JSON(person).XML(person).HTML(buf.String())
	ctx.Negotiate(nil)


```

**Reply by markuswustenberg** [reply]
I'm happy you found a solution. 😊
